linkerd/stack/src/loadshed.rs

```diff
@@ -0,0 +1,85 @@
+use super::layer;
+use futures::{future, TryFutureExt};
+use linkerd_error::Error;
+use std::task::{Context, Poll};
+use thiserror::Error;
+use tower::Service;
+
+/// A middleware that sheds load when the inner `Service` isn't ready.
+#[derive(Debug)]
+pub struct LoadShed<S> {
```

Comment:
We should add some basic tests for this module

Reply:
done [cf8e6d8](https://github.com/linkerd/linkerd2-proxy/pull/2091/commits/cf8e6d85d58ab3748c3fe88b27716dfd09088f79)!